lib/watcher_adapter.js

```diff
@@ -8,9 +8,11 @@ function defaultFilterFunction(name) {
   return /^[^.]/.test(name);
 }
 
-function bindFileEvent(adapter, watcher, event) {
+function bindFileEvent(adapter, watcher, watchedNode, event) {
   watcher.on(event, (filepath, root) => {
     logger.debug(event, root + '/' + filepath);
+    logger.debug(`revise called on node [${watchedNode.id}]`);
+    watchedNode.revise();
```

Comment:
I still don't think you should pass this in. Just register a single change event on the watcher https://github.com/broccolijs/broccoli/pull/397/files#diff-7200ed34a198b0cd92ba001b37c4bc27R37

Reply:
That callback seems to be invoked many many times even on a single file change. Next weeks sync I can show you what I mean but in the mean time we should go with the current solution.